test/machineslib.py

```diff
@@ -408,8 +410,44 @@ def setUp(self):
         # HACK: older c-ws versions always log an assertion, fixed in PR cockpit#16765
         self.allow_journal_messages("json_object_get_string_member: assertion 'node != NULL' failed")
 
+    def downloadVmXml(self, vm):
+        m = self.machine
+
+        m.execute(f"virsh dumpxml {vm} > /var/lib/cockpittest/pkg_specific_artifacts/{vm}.xml")
+        vms_xml = f"/var/lib/cockpittest/pkg_specific_artifacts/{vm}.xml"
+        dest_file = f"{self.label()}-{m.label}-{vm}.xml"
+        attach(vms_xml, move=False)
```

Comment:
Hmm, the first argument is supposed to be a file name, in the local test dir; vms_xml is the file name in the VM. How does that even work like that? You *first* need to m.download() the file and *then* attach it surely?